### ISSUE TITLE
Various suggestions for "Bulk add candidates" system

### ISSUE THREAD

**Issue by jf1** [mysociety/yournextrepresentative#840]
Firstly, instead of ""Bulk add candidates..." in the button text and /bulk_adding/ in the url, I'd describe this as "Add multiple candidates..." in the button text and /add_multiple/ in the url as the adding is restricted to one post.
Ideally a future "Bulk add" facility will be introduced that supports upload of a csv file and importing multiple candidates or candidate data for more than one post.

The first page has 15 forms (to add up to 15 candidates) - is this really necessary?
If I was working on a list with more than 5 candidates I'd be happy to use the page more than once. For all posts with 5 or fewer candidates, the 10+ un-needed forms create a lot of scrolling.

On the second page I'm told "An existing person with this name has been found" - this text should be "An existing person with **a similar** name has been found" - or a pluralised version if we're being really accurate.
![image](https://cloud.githubusercontent.com/assets/4711363/14568233/e1469352-032f-11e6-900f-278374119fc1.png)

Also on this page, I'm forced to select the "Add new person" when there's no existing similar person - this could be auto-selected and I feel it should be.

For extra clarity there could be some text at the top of this page, eg: "Thank you for entering that data. Please check each person below and confirm if they are a new person or if we already know of them from this or a previous election"

**Comment by duncanparkes** [comment]
The number of forms was increased from 10 to 15 because I asked for it. I find it much easier to get each ward right if I'm doing the whole thing in one go and can count the candidates on both sides. I agree there's a lot of scrolling to get to the add button, but perhaps we can solve that in some other way.

**Comment by zarino** [comment]
I wonder whether we could do something like this to add more person forms _only as they’re required_? Might make the page less intimidating.

![form](https://cloud.githubusercontent.com/assets/739624/25744295/fd049086-3190-11e7-889b-267960c135b6.gif)

Source: https://jsfiddle.net/zarino/s5d80vzg/3/

**Comment by zarino** [comment]
I agree that the "An existing person with this name has been found" wording is just plain confusing, when the page then goes on to list a bunch of people with totally different names.

The wording I’d use here is "Did you mean…?" – people are used to that from search engines.

I also wonder whether we can prevent _obviously unrelated_ names from appearing in the list. There is no way that the "Steve Coster" I just typed in is even vaguely related to the "Steve Colborn" and "Steve Cobb" that Elasticsearch is returning. It’s just visual noise. Can we maybe set a threshold somewhere for how similar results must be before we show them?

**Comment by jf1** [comment]
I like the auto-expanding form, but perhaps only add an extra section once there's some text in the field above (rather than expand as soon as it has been clicked on/in).

+1 for "Did you mean...?" - perhaps with a "Please ensure you're not creating a duplicate of an existing candidate" message too?

ps. While this may be beyond this task's scope I hope you're aware that many people think 'bulk upload' means they can upload a correctly-formatted csv. For me, 'bulk input' is a better description of the existing system and clearer than the current 'bulk add' description.

If you are able to look at a csv upload then, especially to address concerns about correctly linking to the correct area and party IDs, I'd suggest one example csv is made available featuring dummy names and area IDs with real party IDs; then allow people to download a custom-generated csv for the election (or post) they're interested in that containing election and post IDs and names but no candidate names, IDs or party IDs so these are all an uploader needs to complete before re-uploading the csv. Party staff and wide-area activists would benefit from this type of feature.
